pkg/util/issutil/leaderselectionpolicy.go

```diff
@@ -6,7 +6,32 @@ SPDX-License-Identifier: Apache-2.0
 
 package issutil
 
-import t "github.com/filecoin-project/mir/pkg/types"
+import (
+	"fmt"
+	"sort"
+	"sync"
+
+	"github.com/fxamacker/cbor/v2"
+
+	t "github.com/filecoin-project/mir/pkg/types"
+)
+
+type LeaderPolicyType uint64
+
+const (
+	Simple LeaderPolicyType = iota
+	Blacklist
+)
+
+var encMode cbor.EncMode
+var once sync.Once
+
+func getEncMode() cbor.EncMode {
+	once.Do(func() {
+		encMode, _ = cbor.CoreDetEncOptions().EncMode()
+	})
+	return encMode
+}
```

Comment:
This also seems like premature optimization. Unlesscbor.CoreDetEncOptions().EncMode() is known to be very expensive, I'd drop this.
If the reason is performance, then it wouldn't even surprise me if this was actually slower, as it needs to access synchronization primitives. Whereas the line being executed inside probably just returns some pre-defined value (although it might be doing some allocations, so I don't know). In any case, this optimization doesn't seem to be worth the complexity it adds.

Reply:
From the official [documentation](https://github.com/fxamacker/cbor):
>  Avoid using init(). For best performance, reuse EncMode and DecMode after creating them.

Reply:
Ok, fair enough!